### PR TITLE
CMA: Improvement for CMA error reporting

### PR DESCRIPTION
**Pull request by shamisp**
Apparently CMA support in Ubuntu 15.10 is broken. For large message sizes it
may end-up transferring only a sub-block of iovec entry, even so it is
not supposed to break it in the middle. This patch adds more detail error
reporting, which helps identify problematic systems.

Signed-off-by: Pavel Shamis (Pasha) <pashareserch@gmail.com>

### CODE REVIEW

**Comment by shamisp** [comment]
@yosefe please take a look

File: src/uct/sm/cma/cma_ep.c [hunk 1]
```diff
@@ -49,8 +50,11 @@ ucs_status_t uct_cma_ep_put_zcopy(uct_ep_h tl_ep, const void *buffer, size_t len
 
     delivered = process_vm_writev(ep->remote_pid, &local_iov, 1, &remote_iov, 1, 0);
     if (ucs_unlikely(delivered != length)) {
+        if (delivered < 0) {
+            ucs_error("process_vm_writev reported an error: %m");
+        }
```
Comment (yosefe) [inline src/uct/sm/cma/cma_ep.c]: maybe should add 'else' here? we need to print just one error message

File: src/uct/sm/cma/cma_ep.c [hunk 2]
```diff
@@ -71,8 +75,11 @@ ucs_status_t uct_cma_ep_get_zcopy(uct_ep_h tl_ep, void *buffer, size_t length,
                                .iov_len = length};
     delivered = process_vm_readv(ep->remote_pid, &local_iov, 1, &remote_iov, 1, 0);
     if (ucs_unlikely(delivered != length)) {
+        if (delivered < 0) {
+            ucs_error("process_vm_readv reported an error: %m");
+        }
```
Comment (yosefe) [inline src/uct/sm/cma/cma_ep.c]: same here - i think we should have an 'else'

File: src/uct/sm/cma/cma_ep.c (hunk 1 above)
Comment (shamisp) [inline src/uct/sm/cma/cma_ep.c]: I prefer to print out two. The information about the size is useful as well.

Comment (yosefe) [inline src/uct/sm/cma/cma_ep.c]: i think 2 messages for same error is confusing, you can print 'length' in your message as well..

Comment (shamisp) [inline src/uct/sm/cma/cma_ep.c]: I think it is just simpler and not really confusing. Just extra line of info. Also, in future if we add while around this loop (which is likely to happen), we don't have to do any modifications. So, I would prefer to leave this way for now...

Comment (yosefe) [inline src/uct/sm/cma/cma_ep.c]: When you add a while we can revise this. For now, we should report errors in a consistent an clear way.

Comment (shamisp) [inline src/uct/sm/cma/cma_ep.c]: It is consistent and clear way. First we return the error, then we return the  status.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/938/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
Any other comments ?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1588/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/981/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
@yosefe - I addressed the comments

**Comment by yosefe** [comment]
:+1:

**Pull request merged**